pmlc/dialect/pxa/ir/ops.cc

```diff
@@ -474,6 +494,54 @@ OpFoldResult PxaReduceOp::fold(ArrayRef<Attribute> cstOperands) {
   return {};
 }
 
+// ---- PxaStoreOp ----
+
+void printPxaStoreOp(OpAsmPrinter &p, PxaStoreOp op) {
+  p << op.getOperation()->getName() << ' ';
+  p << op.val() << ", ";
+  p << op.memref() << '[';
+  auto mapAttr = op->getAttrOfType<AffineMapAttr>(op.getMapAttrName());
+  p.printAffineMapOfSSAIds(mapAttr, op.idxs());
+  p << ']';
+  p.printOptionalAttrDict(op->getAttrs(),
+                          /*elidedAttrs=*/{op.getMapAttrName()});
+  p << " : ";
+  p.printType(op.memref().getType());
+}
+
+// <operation> ::= `pxa.store` ssa-use `,` ssa-use `[` ssa-use-list `]`
+//                 attribute-dict? `:` type
+ParseResult parsePxaStoreOp(OpAsmParser &parser, OperationState &result) {
+  auto indexTy = parser.getBuilder().getIndexType();
+  MemRefType type;
+  AffineMapAttr mapAttr;
+  OpAsmParser::OperandType val, out;
+  SmallVector<OpAsmParser::OperandType, 4> idxs;
+  return failure(
+      parser.parseOperand(val) || parser.parseComma() ||
+      parser.parseOperand(out) ||
+      parser.parseAffineMapOfSSAIds(idxs, mapAttr, "map", result.attributes) ||
+      parser.parseOptionalAttrDict(result.attributes) ||
+      parser.parseColonType(type) || parser.addTypeToList(type, result.types) ||
+      parser.resolveOperand(val, type.getElementType(), result.operands) ||
+      parser.resolveOperand(out, type, result.operands) ||
+      parser.resolveOperands(idxs, indexTy, result.operands));
+}
```

Comment:
We should use the new assembly format support instead of hand writing this: https://mlir.llvm.org/docs/OpDefinitions/#declarative-assembly-format